src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -196,8 +197,27 @@ private static MethodSymbol GetEntryPoint(CSharpCompilation compilation, PEModul
             diagnostics.AddRange(entryPointAndDiagnostics.Diagnostics);
 
             var entryPoint = entryPointAndDiagnostics.MethodSymbol;
-            var synthesizedEntryPoint = entryPoint as SynthesizedEntryPointSymbol;
-            if (((object)synthesizedEntryPoint != null) &&
+
```

Comment:
We should add a comment to the top of the function noting that this returns the real entry point vs. the user defined one.  The name of the function is ambiguous with this feature now.